src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -514,9 +514,26 @@ await diagnosticsWithSameSpan.OrderByDescending(d => d.Severity)
                                           .Select(d => d.Id)
                                           .ToImmutableHashSet();
 
+                // When computing FixAll for unnecessary pragma suppression diagnostic,
+                // we need to include suppressed diagnostics, as well as reported compiler and analyzer diagnostics.
+                // A null value for 'diagnosticIdsForDiagnosticProvider' passed to 'FixAllDiagnosticProvider'
+                // ensures the latter.
+                ImmutableHashSet<string>? diagnosticIdsForDiagnosticProvider;
+                bool includeSuppressedDiagnostics;
+                if (diagnosticIds.Contains(IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId))
+                {
+                    diagnosticIdsForDiagnosticProvider = null;
```

Comment:
It feels weird to have setting this to null create this implicit behavior. Can we refactor this code to make this a less implicit side-effect in the codebase?

Reply:
Yeah, but using null `diagnosticId` is used at multiple places in the IDE diagnostic service in features layer, so I kept it consistent.

https://github.com/dotnet/roslyn/blob/91e67a898f0dacc678a863d612f9d967bf307640/src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs#L65

https://github.com/dotnet/roslyn/blob/91e67a898f0dacc678a863d612f9d967bf307640/src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs#L75-L81

https://github.com/dotnet/roslyn/blob/91e67a898f0dacc678a863d612f9d967bf307640/src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs#L252-L253

I think we need a cleanup across the services to replace null diagnostic IDs with some better representation.